source/server/hot_restart_impl.cc

```diff
@@ -55,7 +55,7 @@ SharedMemory& SharedMemory::initialize(uint64_t stats_set_size, Options& options
 
   int shmem_fd = os_sys_calls.shmOpen(shmem_name.c_str(), flags, S_IRUSR | S_IWUSR);
   if (shmem_fd == -1) {
-    PANIC(fmt::format("cannot open shared memory region {} check user permissions", shmem_name));
+    PANIC(fmt::format("cannot open shared memory region {} check user permissions. Error code: {}", shmem_name, errno));
```

Comment:
Might also consider strerror() here, but not a big deal either way if you have a preference.